packages/core/src/components/widgets/RadioWidget.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { WidgetProps } from "@rjsf/utils";
+
+function RadioWidget<T = any, F = any>({
+  options,
+  value,
+  required,
+  disabled,
+  readonly,
+  autofocus = false,
+  onBlur,
+  onFocus,
+  onChange,
+  id,
+}: WidgetProps<T, F>) {
+  // Generating a unique field name to identify this set of radio buttons
+  const name = Math.random().toString();
+  const { enumOptions, enumDisabled, inline } = options;
+  // checked={checked} has been moved above name={name}, As mentioned in #349;
+  // this is a temporary fix for radio button rendering bug in React, facebook/react#7630.
+  return (
+    <div className="field-radio-group" id={id}>
+      {Array.isArray(enumOptions) &&
+        enumOptions.map((option, i) => {
+          const checked = option.value === value;
+          const itemDisabled =
+            enumDisabled && enumDisabled.indexOf(option.value) != -1;
+          const disabledCls =
+            disabled || itemDisabled || readonly ? "disabled" : "";
+          const radio = (
+            <span>
+              <input
+                type="radio"
```

Comment:
For consistency with the checkboxes component, it probably makes sense to add an `id` for the radio buttons as well.